### ISSUE TITLE
Burp 2.3.4 server error when new fqdn:port syntax is used and CA name isn't left default

### ISSUE THREAD

**Issue by deajan** [grke/burp#808]
Hello,

After playing with burp 2.3.4 earlier, today I updated one of my production servers and a test client to v2.3.4.

Using the following old configuation:
```
mode = client
server = my.burp.server
port = 443
```

Everything runs fine.
But using the new configuration, I get the following:
```
2019-04-02 20:44:23: bin\burp.exe[7824] falling back to 'my.burp.server:4971'
2019-04-02 20:44:23: bin\burp.exe[7824] Connecting to my.burp.server:4971
2019-04-02 20:44:23: bin\burp.exe[7824] Could not connect to my.burp.server:4971
2019-04-02 20:44:23: bin\burp.exe[7824] Failing over
2019-04-02 20:44:23: bin\burp.exe[7824] Connecting to my.burp.server:443
2019-04-02 20:44:23: bin\burp.exe[7824] auth ok
2019-04-02 20:44:23: bin\burp.exe[7824] Server version: 2.3.4
2019-04-02 20:44:23: bin\burp.exe[7824] nocsr ok
2019-04-02 20:44:23: bin\burp.exe[7824] SSL is using cipher: DHE-RSA-AES256-GCM-SHA384 TLSv1.2 Kx=DH       Au=RSA  Enc=AESGCM(256) Mac=AEAD

2019-04-02 20:44:23: bin\burp.exe[7824] cert common name doesn't match configured ssl_peer_cn
2019-04-02 20:44:23: bin\burp.exe[7824] 'burpProdserver'!='my.burp.server:4971'
2019-04-02 20:44:23: bin\burp.exe[7824] check cert failed
```

I indeed changed the **ca_server_name** setting to 'burpProdServer' in the production server config file, which I hadn't in my synthetic burp 2.3.4 tests.
I do understand that since we have multiple paths, the server CA FQDN can't be the server FQDN since the path may change, but then the old school config works without problems.
Any clues please ?

Best regards.

**Comment by grke** [comment]
Hello,

I think that maybe you don't have ssl_peer_cn set in the client burp.conf.

The line: "falling back to 'my.burp.server:4971'" would indicate that, and the line logged previous to that would have said "ssl_peer_cn unset".

The client behaviour in that case is to fall back to using the "server" config.
But I have neglected to strip ":port" from it.

So, a workaround would be to set 'ssl_peer_cn=my.burp.server' in the client burp.conf.

The probable fix would be for me to strip ":port" when guessing the ssl_peer_cn from the "server" config.


What I don't 100% understand is how it works if you don't use the :port syntax.
Have you edited your logs? Did it originally say something like:

'my.burp.server'!='my.burp.server:4971'
or 'burpProdserver'!='burpProdserver:4971'

**Comment by deajan** [comment]
Indeed logs are edited, but my edit is exaclty what I find in ca_server_name.
Ie the log burpProdserver'!='burpProdserver:4971 is exact.

**Comment by deajan** [comment]
Found my problem, which is not 2.3.4 related :)
Indeed, all my clients have 'ssl_peer_cn = burpProdserver' in client config.
Now the stupid thing is that I had to play with batch grep equivalent findstr in order to update my config files while performing the burp upgrade to 2.3.4.

So my grep like looked like ```FINDSTR /I /V "server ^= "```, which also finds the ```ssl_peer_cn = burpProdserver``` line for a shitty batch reason (ie regular expression space character means any in batch !!!!). Welcome to hell.

I've updated my script to ```FINDSTR /I /V /C:"^server = ", the /C: means pass the string as litteral to regex... Which is well...

Anyway, got things sorted out at my side.
The :port remove part is still valid, but sorry for the noise.
